### PR TITLE
Color Rework & Refactor

### PR DESCRIPTION
**Pull request by rossmoody**
Fixes: https://github.com/brave/brave-ui/issues/363

This PR is meant to set the foundations for implementing theme variables more consistently across our components and get us on the road to Light/Dark theme adoption from the base component level. There is also a naming convention consistency refactor component to this which is making the amount of changes look bloated. I wanted to ensure a foundation and naming convention was squared away in one swoop.

Using some of the preliminary work and structure @petemill set up in a `button` component merge many months ago that incorporated a `Theme Provider`, this PR follows the same pattern with some additions to how color is generated and more uniform variable naming conventions. The `toggle` and `button` components were used as initial guinea pigs for implementing the new approach.

image

## High Level Changes
1. Moved `theme` and `color` assets inside `components/style`. This is to future proof the `style` umbrella for future code additions that might not necessarily fit under theme. This is illustrated by the `makeColor` function being broken out into the `function` folder under `style`. In my research I noticed that very often other libraries come to put motion and other style utilities in there.
2. Started `theme` variable foundations for quite a few components. The `button` variables were replaced though follow the same structure (just new names). A lot of the changes from this are the result of consolidating the them variable definitions across components down to the one theme file.
3. Added a systematic color palette generation mechanism. This is in the `style` folder. The naming convention for colors follows the same names and styles. The thing that updated was how they are generated. 
4. Generally I went through and tried to structure file names and folders in a consistent convention. This means making component folder names singular instead of plural and adopting common component names. Examples: `popupsModals` to `modal` | `buttonIndicators` to `button` | `dataTables` to `table` .

## Toggle
The toggle component received a style rework cause it wasn't importing theme variables and also was accommodating for `type?: 'dark' | 'light'` that will be swapped for `theme` color variables. 

## TinyColor
TinyColor is a TS dependency that performs basically the same color functions as Sass or Less. It's an integral part of generating states and consistent alphas in our theme files. It's also what is used to generate the color palette scales. https://github.com/TypeCtrl/tinycolor

## Theming Panel in Storybook
Though the only two components that can actually utilize this capability thus far are `Toggle` and `Button`, storybook has been set up to start implementing theming previews for all base components. This is through an addon that give a toggle to every story for going back and forth between themes. If the component isn't defined for a theme, it will default to `default`.

dark

## Background Colors in Stories
This is another addon that I believe will come to serve a more important purpose as we flush out more components in different themes (especially dark). It allows us to set the background color behind the component to any set of colors we choose. I think this will be a really important part to defining components in dark theme. Getting an honest representation of the components on a dark background is paramount. There are 4 shades of grey right now but we can designate whatever we would like. 

bg

## Favicon 
Brave UI gets a local build favicon and title!

image

## Test plan
TBD

## Integration
- [x] Does this contain changes to src/components or src
- [x] Does this contain changes to src/features for brave-core?

### CODE REVIEW

File: src/components/style/themes/themeInterface.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    toggleBullet: string
+    toggleBulletChecked: string
+    toggleBulletCheckedActive: string
+    toggleBulletFocus: string
+    // Radio
+    radioCircle: string
+    radioStroke: string
+    radioCircleChecked: string
+    radioStrokeChecked: string
+    primaryBackground: string
```
Comment (rossmoody) [inline src/components/style/themes/themeInterface.ts]: `// Modal`

File: src/components/style/themes/default.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    get primaryColorHover () { return new TinyColor(theme.color.primaryColor).makeColor(theme.color.primaryColor, 5) },
+
+    // Accent Color
+    accentColor: colors.blurple500,
+    get accentColorActive () { return new TinyColor(theme.color.accentColor).makeColor(theme.color.accentColor, 7) },
+    get accentColorHover () { return new TinyColor(theme.color.accentColor).makeColor(theme.color.accentColor, 5) },
+
+    // Warn Color
+    warnColor: colors.red500,
+    get warnColorActive () { return new TinyColor(theme.color.warnColor).makeColor(theme.color.warnColor, 7) },
```
Comment (petemill) [inline src/components/style/themes/default.ts]: This pattern means that this color manipulation function is run every time this property is accessed, which means it's running many times unneccessarily. Too many of these could cause jag.

**Comment by petemill** [comment]
Tinycolor should definitely be a dependency through npm `npm install @ctrl/tinycolor` and remove the source from this PR

File: package.json [hunk 3]
```diff
@@ -59,6 +60,7 @@
     "@storybook/addon-centered": "^4.0.7",
     "@storybook/addon-knobs": "^4.0.7",
     "@storybook/addon-options": "^4.0.7",
+    "storybook-addon-styled-component-theme": "^1.1.1",
```
Comment (diracdeltas) [inline package.json]: minor: should this be added alphabetically?

Comment (rossmoody) [reply package.json]: yes, great spot. will fix.

**Comment by petemill** [comment]
This is still WIP, right @rossmoody ?

**Comment by petemill** [comment]
@cezaraugusto as discussed with @rossmoody we're going to try at least a little bit to split this up in to the following list, either commits or PRs as this is a bit concerning in terms of uplift / revert / bug / blame useability

- Add dark theme
- Add storybook theme selection
- Add new color palette and adjust theme variables <-- *the big color one*
- Rename and adjust icons
- Toggle refactor
- Rename some folders (edited)

**Comment by cezaraugusto** [comment]
test

**Comment by cezaraugusto** [comment]
test

**Comment by rossmoody** [comment]
@petemill gonna close this one down and organize from master. i've let things get a little out of hand with updates and it's up to me to break it apart categorically, document and PR again in much smaller increments. Appreciate the previous PRs though.

**Pull request closed without merging**